convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,207 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import glob
+import logging
+import os
+import re
+
+from convert2rhel import actions, exceptions, pkghandler, pkgmanager, utils
+from convert2rhel.systeminfo import system_info
+
+
+loggerinst = logging.getLogger(__name__)
+
+# create classes for installing, removing non rhel kernels, install additional rhel kernel pkgs and update rhel kernel if needed
+class InstallRhelKernel(actions.Action):
+    id = "INSTALL_RHEL_KERNEL"
+
+    def run(self):
+        """Install and update the RHEL kernel."""
+        super(InstallRhelKernel, self).run()
+        loggerinst.info("Installing RHEL kernel ...")
+        output, ret_code = pkgmanager.call_yum_cmd(command="install", args=["kernel"])
+        kernel_update_needed = False
+
+        if ret_code != 0:
+            self.set_result(
+                level="ERROR",
+                id="FAILED_TO_INSTALL_RHEL_KERNEL",
+                title="Failed to install RHEL kernel",
+                description="There was an error while attempting to install the RHEL kernel from yum.",
+                remediations="Please check that you can access the repositories that provide the RHEL kernel.",
+            )
+
+        # Check if kernel with same version is already installed.
+        # Example output from yum and dnf:
+        #  "Package kernel-4.18.0-193.el8.x86_64 is already installed."
+        already_installed = re.search(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
+        if already_installed:
+            rhel_kernel_nevra = already_installed.group(1)
+            non_rhel_kernels = pkghandler.get_installed_pkgs_w_different_fingerprint(
+                system_info.fingerprints_rhel, "kernel"
+            )
+            for non_rhel_kernel in non_rhel_kernels:
+                # We're comparing to NEVRA since that's what yum/dnf prints out
+                if rhel_kernel_nevra == pkghandler.get_pkg_nevra(non_rhel_kernel):
+                    # If the installed kernel is from a third party (non-RHEL) and has the same NEVRA as the one available
+                    # from RHEL repos, it's necessary to install an older version RHEL kernel and the third party one will
+                    # be removed later in the conversion process. It's because yum/dnf is unable to reinstall a kernel.
+
+                    loggerinst.info(
+                        "\nConflict of kernels: One of the installed kernels"
+                        " has the same version as the latest RHEL kernel."
+                    )
+                    self.add_message(
+                        level="INFO",
+                        id="CONFLICT_OF_KERNELS",
+                        description="Conflict of kernels: One of the installed kernels"
+                        " has the same version as the latest RHEL kernel.",
+                    )
```

Comment:
Let's preserve the consistency and extract the description to a variable to be reusable between the log and the add_message